src/components/themeprovider.tsx

```diff
@@ -0,0 +1,8 @@
+"use client";
```

Comment:
I genuinely don't know what this `"use client"` do

Reply:
> I genuinely don't know what this `"use client"` do
It is a very weird keyword that next js 13 app router introduced. If I don't add it, I will not be able to use client hooks like useState, useEffect and the likes. It signifies to the compiler that this component will be dynamic.

Reply:
> I genuinely don't know what this `"use client"` do

It is a very weird keyword that next js 13 app router introduced. If I don't add it, I will not be able to use client hooks like useState, useEffect and the likes. It signifies to the compiler that this component will be dynamic.

Reply:
AHHHHH makes sense.
Thanks for clarifying 🙏🏾

Reply:
@ArinzeJeffrey-droid Additionally, there is also one called 'use server', it is a keyword that tells the compiler that the component is a react server component and therefore, will be able to directly fetch data. TLDR, a react component can become asynchronous and will have the ability to directly access database and do things that are asynchronous.